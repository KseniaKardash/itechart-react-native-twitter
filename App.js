import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View
} from 'react-native';
import Button from './src/Button';

const App = (props: Props) => {
  const { text, onPress } = props;
  return (
    <View style={styles.container}>
      <Text style={styles.welcome}>
        Welcome to React Native!
      </Text>
      <Text style={styles.instructions}>
        To get started, edit App.js
      </Text>
      <Button text={'ENZYME'}/>
    </View>
  );
};
export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
