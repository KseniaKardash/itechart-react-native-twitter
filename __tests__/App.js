import React from "react";
import renderer from "react-test-renderer";
import { shallow, configure } from 'enzyme';
import App from "../App";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("App snapShot", () => {
  it("renders as expected", () => {
    const snap = shallow(<App/>);
    expect(snap).toMatchSnapshot();
  });
});
