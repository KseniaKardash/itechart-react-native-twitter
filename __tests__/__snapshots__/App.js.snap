// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapShot renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          Welcome to React Native!
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          To get started, edit App.js
        </Text>,
        <Button
          text="ENZYME"
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Welcome to React Native!",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Welcome to React Native!",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "To get started, edit App.js",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "To get started, edit App.js",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "ENZYME",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Welcome to React Native!
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "marginBottom": 5,
                "textAlign": "center",
              }
            }
          >
            To get started, edit App.js
          </Text>,
          <Button
            text="ENZYME"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Welcome to React Native!",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Welcome to React Native!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "To get started, edit App.js",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "To get started, edit App.js",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "ENZYME",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
